### PR TITLE
rr/refactor ifdef to ifenabled

### PR DESCRIPTION
**Pull request by riningerr**
## `#ifdef` to `#if ENABLED(<feature>)`

- Changed all `#ifdef PNEUMATICS`, `#ifdef EXT_ADC`, `#ifdef DAC_I2C`,and `#ifdef E_REGULATOR`'s to `#if ENABLED(<feature>)`
- `#ifdef`s are still used, but they refer to defined limits (pressure, temperature, travel, etc.). This convention is used upstream.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -4945,7 +4944,7 @@ inline void gcode_M226() {
   }
 #endif
  
-#ifdef E_REGULATOR
+#if ENABLED(E_REGULATOR)
   /**
    * M236 - Send Value to ADC w/ no EEPROM write
    */
```
Comment (kevingelion) [inline Marlin/Marlin_main.cpp]: I believe M236 has a bunch of pneumatic-related things. Maybe it should also have `&& ENABLED(PNEUMATICS)`

Comment (riningerr) [inline Marlin/Marlin_main.cpp]: I have another branch to take care of that. `E_REGULATOR` will not be defined unless `PNEUMATICS` is defined.

Comment (kevingelion) [inline Marlin/Marlin_main.cpp]: :+1:

**Comment by jminardi** [comment]
LGTM

**Comment by riningerr** [comment]
I am going to test this on Charles before I merge

**Comment by pizzyflavin** [comment]
Reverted last commit. New branch is being made for those changes, as they are unrelated to this branch.

**Comment by pizzyflavin** [comment]
Tested and working on Pru. Got a verbal LGTM from @kdumontnu

**Pull request merged**